src/roman-numerals/index.js

```diff
@@ -4,6 +4,66 @@
  * @param {string} roman The all-caps Roman numeral between 1 and 3999 (inclusive).
  * @returns {number} The decimal equivalent.
  */
-function romanToDecimal(roman) {}
+function romanToDecimal(romanNumeral)
+{
+    //create an object with the basic roman numeral to be referenced
+    const romanDecimal = { I:1, V:5, X:10, L:50, C:100, D:500, M:1000 };
+    
+    //collect the keys i.e the roman letters and store it for reference
+    const romanKeys = Object.keys(romanDecimal);
+    
+	let resultInNumber = 0;//initialize placeholder to hold result as the conversion is  been made.
+	
+    //make sure all user input is in uppercase, if not, convert it to uppercase
+    const romanNumerals = romanNumeral.toUpperCase();
+	
+	for (let letter of romanNumerals)
+	{
+
+		if (romanKeys.indexOf(letter) === -1)
+		{
+            //if any of the letter in the input string parameter is not contained in the roman keys
+            return `invalid roman figure ${letter}`;
+            
+		}
+	}
```

Comment:
Why not do this check concurrently in the loop below.

Reply:
I tried it earlier in the same loop but the program runs in a certain way that doesn't feel right to me. When the invalid roman figure is not at the first index of the user input, the loop still runs to that point before logging the error, but I want the program to terminate irrespective of the previous valid roman figure before it gets to the invalid roman figure within the input string.